### PR TITLE
Implement most common sys.version_info and sys.platform checks

### PR DESCRIPTION
**Pull request by gvanrossum**
Addresses most but not all of #698. (The first two bullets of my plan.)

### CODE REVIEW

File: mypy/semanal.py [hunk 1]
```diff
@@ -2780,6 +2786,139 @@ def infer_if_condition_value(expr: Node, pyversion: Tuple[int, int]) -> int:
...
+                    return None
+                end = index.end_index.value
+            return (begin, end)
+    return None
+
+
+def is_sys_attr(expr: Node, name: str) -> bool:
+    if isinstance(expr, MemberExpr) and expr.name == name:
+        if isinstance(expr.expr, NameExpr) and expr.expr.name == 'sys':
+            # TODO: Guard against a local named sys, etc.
```
Comment (JukkaL) [inline mypy/semanal.py]: Also this won't work with things like `import sys as _sys` -- I found 8 files that did this in the Python 2.7 stdlib.

Comment (JukkaL) [inline mypy/semanal.py]: Another thing that doesn't work is `from sys import platform` (which is arguably questionable style and doesn't seem very common). Not sure if we should support it -- one option is to describe it as unsupported in our docs.

File: test-data/unit/check-unreachable-code.test [hunk 2]
```diff
@@ -111,3 +111,161 @@ else:
...
+foo() + 0
+[builtins fixtures/ops.py]
+[out]
+
+[case testSysVersionInfoSliced8]
+import sys
+if sys.version_info >= (3, 5, 0):
+    def foo() -> int: return 0
+else:
+    def foo() -> str: return ''  # E: All conditional function variants must have identical signatures
```
Comment (JukkaL) [inline test-data/unit/check-unreachable-code.test]: Is this not supported? Maybe add a comment about why.

File: test-data/unit/check-unreachable-code.test [hunk 3]
```diff
@@ -111,3 +111,161 @@ else:
...
+[builtins fixtures/ops.py]
+[out]
+
+[case testSysVersionInfoSliced9]
+import sys
+if sys.version_info[1:] >= (5, 0):
+    def foo() -> int: return 0
+else:
+    def foo() -> str: return ''  # E: All conditional function variants must have identical signatures
```
Comment (JukkaL) [inline test-data/unit/check-unreachable-code.test]: Again, this doesn't work as a reader might expect, so maybe add a comment.

File: test-data/unit/check-unreachable-code.test [hunk 4]
```diff
@@ -111,3 +111,161 @@ else:
...
+    def foo() -> int: return 0
+else:
+    def foo() -> str: return ''
+foo() + 0
+[builtins fixtures/ops.py]
+[out]
+
+[case testSysPlatformNegated]
+import sys
+if not (sys.platform == 'win32'):
```
Comment (JukkaL) [inline test-data/unit/check-unreachable-code.test]: Checks like `if sys.platform.startswith('win')` seem pretty common and wouldn't be hard to support.

**Comment by gvanrossum** [comment]
Thanks for the review! I've added some TODOs and more tests. I'll merge
this once Travis-CI is green.

**Comment by eric-wieser** [comment]
Supporting `sys.version_info.major` would be a nice addition

**Comment by gvanrossum** [comment]
I opened https://github.com/python/mypy/issues/4661 for this -- but not that we'd need buy-in from PyCharm and Google's pytype as well before we can start using this in stubs (where it's most useful).

**Pull request merged**